api/controllers/StorageController.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * Nanocloud turns any traditional software into a cloud solution, without
+ * changing or redeveloping existing source code.
+ *
+ * Copyright (C) 2016 Nanocloud Software
+ *
+ * This file is part of Nanocloud.
+ *
+ * Nanocloud is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Nanocloud is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General
+ * Public License
+ * along with this program.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * UploadController
+ *
+ * @description :: Server-side logic for managing uploads
+ * @help        :: See http://sailsjs.org/#!/documentation/concepts/Controllers
+ */
+
+/* globals AccessToken */
+/* globals PlazaService */
+/* globals Storage */
+/* globals StorageService */
+
+const sha1 = require("sha1");
+
+module.exports = {
+
+  /**
+   * upload a file in storage
+   *
+   * @method uploqd
+   * @public true
+   */
+
+  upload: function(req, res) {
+    let user = req.user;
+
+    StorageService.findOrCreate(user)
+      .then((storage) => {
+        let filename = req.query["filename"];
+
+        req.file(filename).upload({
```

Comment:
Use promise. If `req.file` does not return a promise, use bluebird's `promisify`